### PR TITLE
FDP-47 ~ Updates low voltage measurement message definition.

### PR DESCRIPTION
**Pull request by kevinsmeets**

### CODE REVIEW

File: integration-tests/cucumber-tests-platform-distributionautomation/src/test/resources/features/low-voltage/ProcessLowVoltageMessage.feature [hunk 1]
```diff
@@ -14,13 +14,13 @@ Feature: Low voltage message processing
...
     When MQTT device "TST-01" sends a measurement report
-      | payload | [{"gisnr":"sub-1", "feeder":"1", "D": "02/10/2020 16:03:38", "uts":"1601647418", "data": [0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9,1.0,1.1,1.2,1.3,1.4,1.5,1.6,1.7,1.8,1.9,2.0,2.1,2.2,2.3,2.4,2.5,2.6,2.7,2.8,2.9,3.0,3.1,3.2,3.3,3.4,3.5,3.6,3.7,3.8,3.9,4.0,4.1]}] |
+      | payload | [{"gisnr":"sub-1", "versie":"2", "feeder":"1", "D": "02/10/2020 16:03:38", "uts":"1601647418", "data": [0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9,1.0,1.1,1.2,1.3,1.4,1.5,1.6,1.7,1.8,1.9,2.0,2.1,2.2,2.3,2.4,2.5,2.6,2.7,2.8,2.9,3.0,3.1,3.2,3.3,3.4,3.5,3.6,3.7,3.8,3.9,4.0,4.1,5,6,7,8,9,0,1,2,3,4]}] |
     Then a message is published to Kafka
       | substation identification    | sub-1        |
       | substation name              | substation-1 |
       | bay position                 |            1 |
       | bay identification           | fdr-1        |
-      | numberOfElements             |           41 |
+      | numberOfElements             |           51 |
```
Comment (JelleHoffman) [inline integration-tests/cucumber-tests-platform-distributionautomation/src/test/resources/features/low-voltage/ProcessLowVoltageMessage.feature:23]: Shouldn't this be 52 elements? Including version number?

File: osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/LowVoltageMeasurementToAnalogList.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
       final int length) {
     return (Class<T>)
         Stream.of(LowVoltageMeasurementType.class, LowVoltageMetaMeasurementType.class)
-            .filter(clazz -> clazz.getEnumConstants().length == length)
+            .filter(clazz -> clazz.getEnumConstants().length == length || 41 == length)
```
Comment (JelleHoffman) [inline osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/LowVoltageMeasurementToAnalogList.java:33]: Isn't it better to check for either 41 or 52 elements? Now it is possible to send a new version without being checked.

File: osgp/protocol-adapter-mqtt/osgp-protocol-simulator-mqtt/src/main/resources/osgp-simulator-protocol-mqtt.properties [hunk 3]
```diff
@@ -1,3 +1,3 @@
 mqtt.simulator.spec = mqtt_simulator_peak_shaving_spec.json
 # Start a client publishing the messages from the JSON spec or not
-mqtt.simulator.startClient = false
+mqtt.simulator.startClient = true
```
Comment (JelleHoffman) [inline osgp/protocol-adapter-mqtt/osgp-protocol-simulator-mqtt/src/main/resources/osgp-simulator-protocol-mqtt.properties:3]: Should this be true? Just checking?

**Comment by JelleHoffman** [comment]
It does not seem to me that the code supports both versions. Version 2 is supported but how is version 1 still supported?

File: osgp/protocol-adapter-mqtt/osgp-protocol-simulator-mqtt/src/main/resources/osgp-simulator-protocol-mqtt.properties (hunk 3 above)
Comment (kevinsmeets) [reply osgp/protocol-adapter-mqtt/osgp-protocol-simulator-mqtt/src/main/resources/osgp-simulator-protocol-mqtt.properties]: Restored default value.

File: integration-tests/cucumber-tests-platform-distributionautomation/src/test/resources/features/low-voltage/ProcessLowVoltageMessage.feature (hunk 1 above)
Comment (kevinsmeets) [reply integration-tests/cucumber-tests-platform-distributionautomation/src/test/resources/features/low-voltage/ProcessLowVoltageMessage.feature:24]: Should be the length of the enum containing the measurement fields.

File: osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/LowVoltageMeasurementToAnalogList.java (hunk 2 above)
Comment (kevinsmeets) [reply osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/LowVoltageMeasurementToAnalogList.java]: Implemented a different solution, no longer requiring a check on the hard-coded number.

File: osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/ScadaMeasurementPublishedEventConverter.java [hunk 4]
```diff
@@ -54,7 +54,7 @@ public ScadaMeasurementPublishedEvent convert(
     } catch (final IllegalArgumentException e) {
       LOGGER.error(
           "Measurement values does not have the expected amount of fields. Expecting: {} or {}, actual: {}. Payload: {}.",
-          LowVoltageMeasurementType.values().length,
+          LowVoltageMeasurementTypeVersion1.values().length,
```
[the remark this answers is not in the document]
Comment (kevinsmeets) [reply osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/ScadaMeasurementPublishedEventConverter.java:57]: Version 2 has been added.

**Review by JelleHoffman — APPROVED** [review]

**Pull request merged**